### ISSUE TITLE
CiteFuse SNF function runtime

### ISSUE THREAD

**Issue by shivarice** [SydneyBioX/CiteFuse#3]
Is there anyway to optimize the runtime on CiteFuse()? I'm currently running into issues where it's eating a lot of my RAM and not using any of my cores.

**Comment by sorjuela** [comment]
Hi @YingxinLin,

 I also have this problem with `CiteFuse()`, it took almost 200 GB of memory on my SCE of size: 33694 genes * 37 surface markers * 37292 cells. The object on disk is only 184.7 MB.

Also might be worth implementing some `BiocParallel ` methods to speed it up.

Best,
Stephany

**Comment by xiachenrui** [comment]
Yes, I find the same problem and I find `reducedDimSNF()` can not work on 160,000 cells even I have 1500Gb memory

**Comment by IvanBanBan** [comment]
Has anyone figured out how to bypass this ? or does anyone know an alternative for CITE-seq clustering ? I´ve had to subsample my cells drastically (using geoSketch) for the SNF generation to work without crashing (3000 genes * 7100 cells) .

Performing SNF 
Error: cannot allocate vector of size 2.9 Gb Execution halted
